test/structure/test_structure.py

```diff
@@ -97,166 +103,62 @@ def test_is_idx_subset_non_ch_subset():
-    result = compare_structures(structure1, structure2)
-    assert result == {
-        "left": [res_to_remove2.residue_key],
-        "right": [res_to_remove1.residue_key],
-    }
-
-@pytest.mark.TODO
-def test_merge_right_canonical_only(): #@shaoqz: @imp a better idea is to merge 2 different protein.
-    """Merges differences between two Structure() objects containing only canonical residues."""
-    # TODO(CJ): include non-canonical residues in here
-    TEST_FILE = f"{TEST_DIR}/preparation/data/3NIR.pdb"
-    structure1: Structure = structure_from_pdb(TEST_FILE)
-    structure2: Structure = structure_from_pdb(TEST_FILE)
-    res_to_remove1: Residue = structure2.chains[0][-1]
-    del structure2.chains[0][-1]
-    result = compare_structures(structure1, structure2)
-    assert result == {"left": [res_to_remove1.residue_key], "right": []}
-    result = merge_right(structure1, structure2)
-    assert structure1 == result
-
-@pytest.mark.TODO
-def test_merge_right_with_ligand():
-    """Merges differences between two Structure() objects containing a Ligand() object."""
-    TEST_FILE = f"{DATA_DIR}/FAcD-FA-ASP_rmW.pdb"
-    structure1: Structure = structure_from_pdb(TEST_FILE)
-    structure2: Structure = structure_from_pdb(TEST_FILE)
-    structure2.remove_chain("B")
-    assert structure1 != structure2
-    structure2: Structure = merge_right(structure1, structure2)
-    assert structure1 == structure2
-    print(DATA_DIR)
-    # assert False
-
-@pytest.mark.TODO
-def test_merge_right_with_metal_atom():
-    """Merges differences between two Structure() objects containing a MetalUnit() object."""
-    TEST_FILE = f"{DATA_DIR}/1NVG.pdb"
-    structure1: Structure = structure_from_pdb(TEST_FILE)
-    structure2: Structure = structure_from_pdb(TEST_FILE)
-    # NOTE(CJ): Below is NOT the recommended way to manipulate this stuff
-    del structure2._chains[0]._residues[-1]
-    print(structure1._chains[0]._residues)
-    print(structure2._chains[0]._residues)
-    assert structure1 != structure2
-    structure2: Structure = merge_right(structure1, structure2)
-    assert structure1 == structure2
-
-@pytest.mark.TODO
 def test_round_trip_pdb():
     """Ensuring that the Structure() class be loaded into a .pdb and saved back in a round trip without error."""
     # FIXME(CJ): This test doesn"t currently work for 1NVG: figure out the PDBline stuff
     # that will make this work
     TEST_FILE = f"{DATA_DIR}/1NVG.pdb"
     actual_file = f"{DATA_DIR}/1NVG_cpy.pdb"
-    structure1: Structure = structure_from_pdb(TEST_FILE, "all")
+    structure1: Structure = PDBParser.get_structure(TEST_FILE, "all")
     fs.safe_rm(actual_file)
     assert not os.path.exists(actual_file)
-    structure1.to_pdb(actual_file)
+    #structure1.to_pdb(actual_file)
+    structure1_contents: List[str] = PDBParser.get_file_str(
+        structure1).splitlines()
+    fs.write_lines(actual_file, structure1_contents)
     assert os.path.exists(actual_file)
```

Comment:
shouldn't this be in the test_pdb_io.py and there are several similar tests there

Reply:
Good idea. `to_pdb()` used to be a part of `Structure`. Moved.